### PR TITLE
Cut(eos_designs): Remove deprecated key short_esi

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Remove deprecated key adapter_config.port_channel.short_esi

## Related Issue(s)

Fixes #

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Remove deprecated key adapter_config.port_channel.short_esi and code associated with it.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by MaheshGSLAB — APPROVED** [review]

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ The following data model keys have been removed from `eos_designs` in v5.0.0.
 | Removed key | New key |
 | ----------- | ------- |
 | old key 1(cvp_instance_ip) | new key(TODO) |
-| old key 2(defs_adapter_config) | new key(TODO) |
+| adapter_config.port_channel.short_esi | adapter_config.port_channel.ethernet_segment.short_esi |
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: I think, the respective data-models should be mentioned here, like -
`network_ports[].short_esi | network_ports[].ethernet_segment.short_esi`

same for `connected_endpoints` and `port_profiles`

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: @ClausHolbechArista

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: Yes please be elaborate so someone searching will find it.

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/defs_adapter_config.schema.yml [hunk 2]
```diff
@@ -474,13 +474,6 @@ $defs:
               When MPASS is enabled on an MLAG port-channel, MLAG peers coordinate to function as a single PTP logical device.
               Arista PTP enabled devices always place PTP messages on the same physical link within the port-channel.
               Hence, MPASS is needed only on MLAG port-channels connected to non-Arista devices.
-          short_esi:
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/defs_adapter_config.schema.yml:477]: Please also clean up some old draft starting in line 434

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: You also missed the port_channel part here.
```suggestion
| network_ports[].port_channel.short_esi | network_ports[].ethernet_segment.short_esi |
| <connected_endpoints_key>[].adapters[].port_channel.short_esi | <connected_endpoints_key>[].adapters[].ethernet_segment.short_esi |
| port_profiles[].port_channel.short_esi | port_profiles[].ethernet_segment.short_esi |
```

File: python-avd/pyavd/_eos_designs/structured_config/connected_endpoints/utils.py [hunk 3]
```diff
@@ -102,11 +102,8 @@ def _get_short_esi(
             return None
 
         # short_esi is only set when called from sub-interface port-channels.
```
Comment (laxmikantchintakindi) [inline python-avd/pyavd/_eos_designs/structured_config/connected_endpoints/utils.py:104]:
```suggestion
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md (hunk 1 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: Fixed

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/defs_adapter_config.schema.yml (hunk 2 above)
Comment (Vibhu-gslab) [reply python-avd/pyavd/_eos_designs/schema/schema_fragments/defs_adapter_config.schema.yml:477]: Removed!

**Review by Shivani-gslab — APPROVED** [review]
LGTM!

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**